### PR TITLE
Replace print() statements with logging.info() calls for better logging

### PR DESCRIPTION
**Pull request by Am0stafa**
## Description
Use logging.info() instead of print() statements for more flexible logging

## Motivation and Context
Logging is a more flexible and customizable way to display information in the project, as compared to using `print()` statements and you can easily switch between different log levels, such as debug, info, warning, error, and critical by specifying using -l argument.

## How to Test
This change should not affect the functionality of the project. To test, simply run the project and verify that the expected log messages are displayed in the console or log file.

## Related Issues
None.

## Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Checklist
- [x] I have tested this change thoroughly.
- [x] I have added appropriate comments to the code for future reference.
- [x] I have updated the documentation (if applicable).

### CODE REVIEW

File: src/gcp_scanner/credsdb.py [hunk 1]
```diff
@@ -189,12 +189,12 @@ def find_creds(explicit_path: Optional[str] = None) -> List[str]:
         search_paths.append(full_path)
 
   for dir_path in search_paths:
-    print(f"Scanning {dir_path} for credentials.db")
+    logging.info(f"Scanning {dir_path} for credentials.db")
```
Comment (ZetaTwo) [inline src/gcp_scanner/credsdb.py:192]: For log calls, don't use f-strings, instead do `logging.info("Scanning %s for credentials.db", dir_path)`

File: src/gcp_scanner/credsdb.py [hunk 2]
```diff
@@ -189,12 +189,12 @@ def find_creds(explicit_path: Optional[str] = None) -> List[str]:
...
     full_path = os.path.join(dir_path, "credentials.db")
     if os.path.exists(full_path) and os.access(full_path, os.R_OK):
-      print(f"Identified accessible gcloud config profile {full_path}")
+      logging.info(f"Identified accessible gcloud config profile {full_path}")
```
Comment (ZetaTwo) [inline src/gcp_scanner/credsdb.py:195]: For log calls, don't use f-strings.

File: src/gcp_scanner/credsdb.py [hunk 3]
```diff
@@ -189,12 +189,12 @@ def find_creds(explicit_path: Optional[str] = None) -> List[str]:
...
       list_of_creds_files.append(full_path)
-  print(f"Identified {len(list_of_creds_files)} credential DBs")
+  logging.info(f"Identified {len(list_of_creds_files)} credential DBs")
```
Comment (ZetaTwo) [inline src/gcp_scanner/credsdb.py:197]: For log calls, don't use f-strings.

File: src/gcp_scanner/credsdb.py [hunk 4]
```diff
@@ -263,7 +263,7 @@ def extract_creds(path_to_creds_db: str) -> List[Tuple[str, str, str]]:
       logging.info("Found valid access token for %s", row[0])
       access_token = access_tokens[row[0]]
     res.append(SA(row[0], row[1], access_token))
-  print(f"Identified {len(res)} credential entries")
+  logging.info(f"Identified {len(res)} credential entries")
```
Comment (ZetaTwo) [inline src/gcp_scanner/credsdb.py:266]: For log calls, don't use f-strings.

File: src/gcp_scanner/scanner.py [hunk 5]
```diff
@@ -93,7 +93,7 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
 
       project_id = project['projectId']
       project_number = project['projectNumber']
-      print(f'Inspecting project {project_id}')
+      logging.info(f'Inspecting project {project_id}')
```
Comment (ZetaTwo) [inline src/gcp_scanner/scanner.py]: For log calls, don't use f-strings.

**Review by ZetaTwo — CHANGES_REQUESTED** [review]
Overall fine. Just need to replace f-strings with the interpolation provided by the log library.

I also wonder if we should set up logging more properly and create a log object with settings set on it but that is a separate issue.

**Comment by Am0stafa** [comment]
@ZetaTwo yeah as I was reading through the logging documentation I thought about that.
what about now i add a configuration to add the logs to a logging file ?

**Comment by mshudrak** [comment]
Ideally, we need several levels of logging. I left those messages via print on purpose in order to guide user of what's going on during the GCP Scanner run. We need them to be printed by default (without -l INFO) option.

**Comment by Am0stafa** [comment]
@mshudrak yeah that makes sense but what we can do is make the logging level by default at the information level such that even if the user didn't specify the logging level, the user still would be able to see the logs

**Comment by mshudrak** [comment]
@Am0stafa yes it can work but now we have too much information in the console from all logging which we don't want to have by default :) I'd suggest at least one more level of logging for those messages that were print().

**Comment by Am0stafa** [comment]
@mshudrak yeah I get your point I thought that would be a meaningful addition. Now I adjusted the formatting so that the message looks better at the console and is easier to read, along with the ability to disable all logs.

**Comment by Am0stafa** [comment]
Firstly sorry for all the pylint fixes, it took me a while to understand it and configure it,
Secondly, @mshudrak Do you think I should close this PR if you think it's better to print rather than show all the logging form information levels and below or disable them?

**Comment by mshudrak** [comment]
@Am0stafa if you can implement additional level of logging in this PR it is fine to keep it.

**Comment by Am0stafa** [comment]
@mshudrak actually its better if it's a print statement rather than adding another level of logging,
so what I will do is close this PR and make another one where I only change the logging format.

**Pull request closed without merging**